samples/python/aruco_detect_board_charuco.py

```diff
@@ -0,0 +1,108 @@
+-H, --help - show help
+"""
+
+import argparse
+import numpy as np
+import cv2 as cv
+
+
+def read_camera_parameters(filename):
+    fs = cv.FileStorage(filename, cv.FileStorage_READ)
+    if fs.isOpened():
+        cam_matrix = fs.getNode("camera_matrix").mat()
+        dist_coefficients = fs.getNode("distortion_coefficients").mat()
+        return True, cam_matrix, dist_coefficients
+    return False, [], []
+
+
+def main():
+    # parse command line options
+    parser = argparse.ArgumentParser(description="detect markers and corners of charuco board, estimate pose of charuco"
+                                     "board", add_help=False)
+    parser.add_argument("-H", "--help", help="show help", action="store_true", dest="show_help")
+    parser.add_argument("-v", "--video", help="Input from video or image file, if omitted, input comes from camera",
+                        default="", action="store", dest="v")
+    parser.add_argument("-w", help="Number of squares in X direction", default="3", action="store", dest="w", type=int)
+    parser.add_argument("-h", help="Number of squares in Y direction", default="3", action="store", dest="h", type=int)
+    parser.add_argument("-sl", help="Square side length", default="1.", action="store", dest="sl", type=float)
+    parser.add_argument("-ml", help="Marker side length", default="0.5", action="store", dest="ml", type=float)
+    parser.add_argument("-d", help="dictionary: DICT_4X4_50=0, DICT_4X4_100=1, DICT_4X4_250=2,  DICT_4X4_1000=3,"
+                                   "DICT_5X5_50=4, DICT_5X5_100=5, DICT_5X5_250=6, DICT_5X5_1000=7, DICT_6X6_50=8,"
+                                   "DICT_6X6_100=9, DICT_6X6_250=10, DICT_6X6_1000=11, DICT_7X7_50=12, DICT_7X7_100=13,"
+                                   "DICT_7X7_250=14, DICT_7X7_1000=15, DICT_ARUCO_ORIGINAL = 16}",
+                        default="0", action="store", dest="d", type=int)
+    parser.add_argument("-ci", help="Camera id if input doesnt come from video (-v)", default="0", action="store",
+                        dest="ci", type=int)
+    parser.add_argument("-c", help="Input file with calibrated camera parameters", default="", action="store", dest="c")
+
+    args = parser.parse_args()
+
+    show_help = args.show_help
+    if show_help:
+        parser.print_help()
+        return
+    w = args.w
+    h = args.h
+    sl = args.sl
+    ml = args.ml
+    d = args.d
+    video = args.v
+    ci = args.ci
+
+    c = args.c
+    cam_matrix = []
+    dist_coefficients = []
+    if c != "":
+        _, cam_matrix, dist_coefficients = read_camera_parameters(c)
+
+    aruco_dict = cv.aruco.getPredefinedDictionary(d)
+    board_size = (w, h)
+    board = cv.aruco.CharucoBoard(board_size, sl, ml, aruco_dict)
+    charuco_detector = cv.aruco.CharucoDetector(board)
+
+    input_video = cv.VideoCapture()
+    wait_time = 0
+    if video != "":
+        input_video.open(video)
+    else:
+        input_video.open(ci)
+        wait_time = 10
```

Comment:
> wait_time

What is the purpose of this?

`waitKey(0)` will wait forever - we don't really want that for input video file.

---

What do we have in the other samples?

Reply:
Branch with `wait_time=0` works only for images. `wait_time=10` in video/camera input case.
```
    wait_time = 0
    if video != "":
        input_video.open(video)
    else:
        input_video.open(ci)
        wait_time = 10
```

Reply:
`wait_time=0` is useful for viewing images. Is this parameter a big problem for working with video?

Reply:
`cv.waitKey(0)` uses in qrcode.py sample: https://github.com/opencv/opencv/blob/4.x/samples/python/qrcode.py

Reply:
removed `waitKey(0)` from loop